torch/ao/quantization/fx/_lower_to_native_backend.py

```diff
@@ -371,77 +380,61 @@ def _lower_weighted_ref_functional(
     """
     Traverse the graph and replace functional reference patterns with their quantized versions.
     """
-    for ref_func, (q_func, q_relu_func) in LOWER_FUNCTIONAL_MAP.items():
-        configurations = itertools.product(
-            (False, True),  # is_relu: whether ref_func is wrapped in a relu op
-            (False, True),  # has_bias: whether bias is passed as an extra argument to ref_func
-        )
-        for is_relu, has_bias in configurations:
-            if is_relu and q_relu_func is None:
-                continue
-
-            # Set up match pattern: (dequantize - [relu_op - ] func_op - quantize)
-            # Func args: (dequantized inputs, dequantized weights[, bias])
-            # Quantize args: (func, scale, zp, dtype)
-            func_pattern: Tuple[Any, ...] = ()
-            if has_bias:
-                func_pattern = (ref_func, "dequantize", "dequantize", MatchAllNode)
-            else:
-                func_pattern = (ref_func, "dequantize", "dequantize")
-            if is_relu:
-                func_pattern = (F.relu, func_pattern)
-            pattern = (torch.quantize_per_tensor, func_pattern, MatchAllNode, MatchAllNode, MatchAllNode)
-
-            # Iterate through nodes in the graph to find a match
-            # If there is a match, replace the above pattern with the corresponding quantized op
-            modules = dict(model.named_modules(remove_duplicate=False))
-            nodes = list(model.graph.nodes)
-            for n in model.graph.nodes:
-                if not is_match(modules, n, pattern):
-                    continue
-                q_node = n
-                (func_node, output_scale_node, output_zp_node, dtype) = q_node.args
-                if is_relu:
-                    relu_node = func_node
-                    func_node = relu_node.args[0]
-                else:
-                    relu_node = None
-                input_dq_node = func_node.args[0]
-                weight_dq_node = func_node.args[1]
-
-                if should_skip_lowering(func_node, qconfig_map):
-                    continue
+    modules = dict(model.named_modules(remove_duplicate=False))
+    nodes = list(model.graph.nodes)
+    for n in model.graph.nodes:
+        # Step 0: Find nodes that match this pattern (dequantize - functional op - quantize)
+        # We search for the pattern backwards, starting with the quantize node
+        # Quantize node args: (func, scale, zp, dtype)
+        if n.target != torch.quantize_per_tensor:
+            continue
+        q_node = n
+        (func_node, output_scale_node, output_zp_node, _) = q_node.args
+        if func_node.op != "call_function":
+            continue
+        # Handle cases where the functional op is wrapped in a ReLU
+        if func_node.target == F.relu:
+            relu_node = func_node
+            func_node = relu_node.args[0]
+        else:
+            relu_node = None
+        if func_node.target not in LOWER_FUNCTIONAL_MAP:
```

Comment:
will need to make sure func_node.op == "call_function" here as well

Reply:
Actually I check for that in L393, but I can merge the two checks